src/deepsparse/utils/onnx.py

```diff
@@ -68,6 +68,49 @@
 }
 
 
+def save_onnx_to_path(model: Model, model_path: str, external_data_file: str) -> bool:
+    """
+    Save model to the given path.  If the model has external data, store the
+    external data in 'external_data_file'.
+    Returns False if the model had no external data, True otherwise.
+    """
+    try:
+        onnx.checker.check_model(model)
+        onnx.save(model, model_path)
+        return False
+    except ValueError:
+        # Presumably the checker has failed because of external data
+        onnx.save_model(
+            model,
+            model_path,
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+            location=external_data_file,
+        )
+        return True
+
+
+@contextlib.contextmanager
+def save_onnx(model: Model) -> str:
```

Comment:
would rename to specify that the functionality added here is to save to temporary files. would also rename `save_onnx_to_path` as `save_onnx` so that can be our source of truth for saving in general